crates/bevy_scene/src/dynamic_scene.rs

```diff
@@ -35,7 +36,28 @@ impl DynamicScene {
         let mut scene = DynamicScene::default();
         let type_registry = type_registry.read();
 
-        for archetype in world.archetypes().iter() {
+        let mut archetypes = world.archetypes().iter();
+
+        // Empty archetype
+        let _ = archetypes.next().unwrap();
+
+        // Resources archetype
+        let resources_archetype = archetypes.next().unwrap();
+        for component_id in resources_archetype.components() {
+            let reflect_resource = world
+                .components()
+                .get_info(component_id)
+                .and_then(|info| type_registry.get(info.type_id().unwrap()))
+                .and_then(|registration| registration.data::<ReflectResource>());
+            if let Some(reflect_resource) = reflect_resource {
+                if let Some(resource) = reflect_resource.reflect_resource(world) {
+                    scene.resources.push(resource.clone_value());
+                }
+            }
+        }
+
+        // Other archetypes
+        for archetype in archetypes {
```

Comment:
IMO this should have a comment calling out that we need to ensure that we iterate first. Iteration being stateful is a bit surprising at times, and swapping this code with the code above feels like it should be a harmless code re-organization change when it very much isn't.

Reply:
Fixed in f43d667